flaml/model.py

```diff
@@ -1965,6 +1965,189 @@ class XGBoostLimitDepth_TS(TS_SKLearn):
+            search_space["optimization.batch_size"] = self.fix_args.get("batch_size", 128)
+            for key, value in self.params.items():
+                if key in TUNABLE_HP:
+                    # NOTE: FLAML uses np.float64 but AG uses float, need to transform
+                    if isinstance(value, np.float64):
+                        search_space[key] = value.item()
+                    else:
+                        search_space[key] = value
+            return cfg
+        
+        else:
+            raise ValueError("the pytorch automm model is not supported. ")
+            # from autogluon.text.text_prediction.presets import get_text_preset
+
+            # cfg, overrides = get_text_preset("default")  # get preset for text+num+cat+fusion
+            # # TODO: set the search space for the auto_mm in AG 0.4.0
+            # cfg.hf_text.checkpoint_name = self.fix_args["hf_text.checkpoint_name"]
+            # # get search configs from self.params and set here
+            # TUNABLE_HP = []
+            # for key, value in self.params.items():
+            #     if key in TUNABLE_HP:
+            #         # NOTE: FLAML uses np.float64 but AG uses float, might need to transform
+            #         if isinstance(value, np.float64):
+            #             search_space[key] = value.item()
+            #         else:
+            #             search_space[key] = value
+            return cfg
+            
+
+    def _set_seed(self, seed):
+        import random
+        import mxnet as mx
+        import torch as th
+        th.manual_seed(seed)
+        mx.random.seed(seed)
+        np.random.seed(seed)
+        random.seed(seed)
+
+    def fit(self, X_train=None, y_train=None, budget=None, **kwargs):
+        self._kwargs = kwargs
+        self._init_fix_args(kwargs)
+        # the seed set in the bash script for ag experiment is 123
+        seed = self.params.get("seed", 123)
+        self._set_seed(seed)
+
+        # get backbone and fusion strategy
+        text_backbone = self.fix_args["text_backbone"]
+        multimodal_fusion_strategy = self.fix_args["multimodal_fusion_strategy"]
+
+        # get & set the save dir, get the dataset info
+        save_dir = self.fix_args["output_dir"]
+        label_column = self.fix_args["label_column"]
+        dataset_name = self.fix_args["dataset_name"]
+        ag_model_save_dir = os.path.join(save_dir, f"{dataset_name}_ag_text_multimodal_{text_backbone}\
+                                                    _{multimodal_fusion_strategy}_no_ensemble")
+
+        # set the hyperparameters
+        self.hyperparameters = self._init_hp_config(text_backbone, multimodal_fusion_strategy)
+        PROBLEM_TYPE_MAPPING = {"mm_binary": "binary", "mm_multi": "multiclass", "mm_regression": "regression"}
+        TASK_METRIC_MAPPING = {"mm_multi": "acc", "mm_binary": "roc_auc", "mm_regression": "r2"}

```

Comment:
Please implement the metric mapping in flaml/nlp/utils.py:load_default_huggingface_metric_for_task